IdentityCore/src/oauth2/token/MSIDAccessToken.m

```diff
@@ -229,6 +229,24 @@ - (void)fillExtendedExpiryFromResponse:(MSIDTokenResponse *)response
     }
 }
 
+- (void)fillTargetFromResponse:(MSIDTokenResponse *)response
+                 requestParams:(MSIDRequestParameters *)requestParams
+{
+    // Because resource/scopes is not always returned in the token response, we rely on the input resource/scopes as a fallback
+    _target = response.target ? response.target : requestParams.target;
+    
+    // ".default" scope in request should be manually added to access token
+    if ([response isKindOfClass:[MSIDAADV2TokenResponse class]])
+    {
+        if (requestParams.scopes.count == 1 && [requestParams.scopes.firstObject.lowercaseString hasSuffix:@".default"])
```

Comment:
I suggest we move this logic to MSIDRequestParameters to have them decide if they should add additional scopes to the list, e.g. something like additionalScopes property or method which returns only if ".default" was there (just in case we have to add additional cases like that in future).

Reply:
Nice idea! Let me change it :)